### PR TITLE
Add Contribution Guidelines

### PR DESCRIPTION
**Pull request by joefitzgerald**
The title says it all. This PR seeks to make the implicit explicit. It borrows heavily from the Atom contribution guidelines, making minor changes for linter (like requiring PRs to not have lint errors :wink:).

### CODE REVIEW

**Comment by steelbrain** [comment]
LGTM.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  [Jasmine](http://jasmine.github.io/) specs in the `./spec` folder.
+- treat `describe` as a noun or situation.
+- treat `it` as a statement about state or how an operation changes state.
+
+### Example
+
+```coffee
+describe 'a dog', ->
+ it 'barks', ->
+ # spec here
```
Comment (AsaAyers) [inline CONTRIBUTING.md]: The `# spec here` comments could use one more level of indentation

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+* Perform a [cursory search](https://github.com/atom-community/linter/search?q=&type=Issues&utf8=%E2%9C%93)
+  to see if a similar issue has already been submitted.
+* Please setup a [profile picture](https://help.github.com/articles/how-do-i-set-up-my-profile-picture)
+  to make yourself recognizable and so we can all get to know each other better.
+
+## Pull Requests
+
+* We prefer small, focused, single-responsibility pull requests that include tests where possible. These can be contrasted with large pull requests, pull requests with multiple unrelated concerns, and pull requests which have no tests.
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* **Please ensure that your pull request has no lint errors.** This is a project for linters after all,
```
Comment (iam4x) [inline CONTRIBUTING.md]: Let me create a new PR to fix code-style before.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  such as Emacs, vi, Xcode, etc.
+* Check the dev tools (`alt-cmd-i`) for errors to include. If the dev tools
+  are open _before_ the error is triggered, a full stack trace for the error
+  will be logged. If you can reproduce the error, use this approach to get the
+  full stack trace and include it in the issue.
+* On Mac, check Console.app for stack traces to include if reporting a crash.
+* Perform a [cursory search](https://github.com/atom-community/linter/search?q=&type=Issues&utf8=%E2%9C%93)
+  to see if a similar issue has already been submitted.
+* Please setup a [profile picture](https://help.github.com/articles/how-do-i-set-up-my-profile-picture)
+  to make yourself recognizable and so we can all get to know each other better.
```
Comment (k2b6s9j) [inline CONTRIBUTING.md]: I actually really love this. It's a lot easier talking to someone who I can associate with a face/emblem.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+* Include screenshots and animated GIFs in your pull request whenever possible.
+* **Please ensure that your pull request has no lint errors.** This is a project for linters after all,
+  so please ensure you have the [linter-coffeelint](https://atom.io/packages/linter-coffeelint) and [linter-eslint](https://atom.io/packages/linter-eslint) packages installed in
+  Atom.
+* Include thoughtfully-worded, well-structured
+  [Jasmine](http://jasmine.github.io/) specs in the `./spec` folder. Run them using `apm test`. See
+  the [Specs Styleguide](#specs-styleguide) below.
+* Document new code based on the
+  [Documentation Styleguide](#documentation-styleguide)
+* End files with a newline.
```
Comment (k2b6s9j) [inline CONTRIBUTING.md]: It might be worth noting that both Atom and the GitHub Online Editor do this for you.

**Comment by k2b6s9j** [comment]
My god this is great. :+1:

I might start stealing this for other projects. :grinning:

File: CONTRIBUTING.md (hunk 1 above)
Comment (joefitzgerald) [inline CONTRIBUTING.md]: Fixed in 212535a25c0fffd4515e65dde4765bdc5ba1708b.

**Pull request merged**